lib/artist.rb

```diff
@@ -0,0 +1,27 @@
+class Artist
+  attr_accessor :name
+  attr_reader :songs
+  extend Concerns::Findable
+  extend Concerns::Listable
+  extend Concerns::Nameable
+  extend Concerns::Persistable::ClassMethods
+  include Concerns::Persistable::InstanceMethods
+
+  def initialize(name)
+    @name = name
+    @songs = []
+  end
+
+  def add_song(song)
+    song.artist = self unless song.artist
+    @songs << song unless @songs.include? song
+  end
+
+  def genres
+    songs.map(&:genre).uniq
+  end
+
+  def to_s
+    @name
```

Comment:
Preference? Or was this needed somewhere?

Reply:
It made the list_by_attributes much cleaner, rather than having to call Arist.name or Genre.name or something like that if I wanted to print an artist.
It should work generically

https://github.com/learn-co-students/ruby-music-library-cli-pca-001/blob/79e048d72167185666e0dbd9334c9a287f0bf0c2/lib/concerns/listable.rb#L22

Reply:
`markdown`

Reply:
Nice!